File: benchmarks/go/06_prime_sieve.go
// Prime Sieve Benchmark - Go Implementation
// Sieve of Eratosthenes algorithm

package main

import "fmt"

func main() {
	// Benchmark parameter - find primes up to n
	n := 10000

	// Initialize sieve (true = potentially prime)
	sieve := make([]bool, n+1)
	for i := range sieve {
		sieve[i] = true
	}
	sieve[0] = false
	sieve[1] = false

	// Sieve of Eratosthenes
	for p := 2; p*p <= n; p++ {
		if sieve[p] {
			// Mark all multiples of p as not prime
			for i := p * p; i <= n; i += p {
				sieve[i] = false
			}
		}
	}

	// Count primes
	count := 0
	for _, isPrime := range sieve {
		if isPrime {
			count++
		}
	}

	fmt.Printf("primes up to %d: %d\n", n, count)
}

File: benchmarks/rust/06_prime_sieve.rs
// Prime Sieve Benchmark - Rust Implementation
// Sieve of Eratosthenes algorithm

fn main() {
    // Benchmark parameter - find primes up to n
    let n = 10000;
    
    // Initialize sieve (true = potentially prime)
    let mut sieve = vec![true; n + 1];
    sieve[0] = false;
    sieve[1] = false;
    
    // Sieve of Eratosthenes
    let mut p = 2;
    while p * p <= n {
        if sieve[p] {
            // Mark all multiples of p as not prime
            let mut i = p * p;
            while i <= n {
                sieve[i] = false;
                i += p;
            }
        }
        p += 1;
    }
    
    // Count primes
    let count = sieve.iter().filter(|&&is_prime| is_prime).count();
    println!("primes up to {}: {}", n, count);
}
